### ISSUE TITLE
Support fetching posts using URL on Soapbox (Pleroma)

### ISSUE THREAD

**Issue by bkil** [friendica/friendica#12323]
Adding contacts based on user@host works, but https://example.invalid/@user does not.

> The network type couldn't be detected. Contact can't be added.

**Comment by bkil** [comment]
Neither can I pull down individual posts to reply - it tells me there are no results.

**Comment by annando** [comment]
The format `https://example.invalid/@user` is not valid for Pleroma. Also - sadly - the URL that Pleroma uses to display posts does not support being fetched by ActivityPub.

**Comment by bkil** [comment]
Yep, I found it out later that this also works: https://example.invalid/users/user Do you happen to know any workaround to fetch the link by peeking into the HTML perhaps?

**Comment by bkil** [comment]
Friendica could successfully fetch it through the URL found
* within the `<id>` tag of each post in the feed https://example.invalid/users/USER/feed.atom
* under the `uri` key of the JSON returned by  https://example.invalid/api/v1/statuses/ID and https://example.invalid/api/v1/statuses/?ids[]=ID

**Comment by MrPetovan** [comment]
Yes, it could, but how do we arrive there from a post URL?

**Comment by bkil** [comment]
```
function get_pleroma_api_uri(string $url): string {
  if (preg_match('/^([^:]*:\/\/[^\/]+)\/notice\/([0-9A-Za-z]{18})$/', $url, $match)) {
    return $match[1] . '/api/v1/statuses/' . $match[2];
  } else {
    return null;
  }
}
```
